elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -1689,4 +1754,40 @@ public InjectedValue<PathManager> getPathManagerInjector() {
         }
     }
 
+    private static TrustManagerFactory createTrustManagerFactory(Provider[] providers, String providerName, String algorithm) throws StartException {
+        TrustManagerFactory trustManagerFactory = null;
+
+        if (providers != null) {
+            for (Provider current : providers) {
+                if (providerName == null || providerName.equals(current.getName())) {
+                    try {
+                        // TODO - We could check the Services within each Provider to check there is one of the required type/algorithm
+                        // However the same loop would need to remain as it is still possible a specific provider can't create it.
+                        return TrustManagerFactory.getInstance(algorithm, current);
+                    } catch (NoSuchAlgorithmException ignored) {
+                    }
+                }
+            }
+            if (trustManagerFactory == null)
+                throw ROOT_LOGGER.unableToCreateManagerFactory(TrustManagerFactory.class.getSimpleName(), algorithm);
+        }
+
+        try {
+            return TrustManagerFactory.getInstance(algorithm);
+        } catch (NoSuchAlgorithmException e) {
+            throw new StartException(e);
+        }
+    }
+
+    public static KeyStore getKeyStoreFromTrustManager(X509ExtendedTrustManager trustManager) throws Exception {
+        // TODO: proporly extract the keystore from the trustmanager
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        trustStore.load(null, null);
+        X509Certificate[] trustedCerts = trustManager.getAcceptedIssuers();
+        for (X509Certificate certificate : trustedCerts) {
+            trustStore.setCertificateEntry(certificate.getSerialNumber().toString(), certificate);
+        }
+        return trustStore;
+    }
```

Comment:
I don't think this is valid. 
I could not find a more valid way of getting the truststore from a trustmanager, which is needed to build a X509RevocationTrustManager.